src/http/one/FieldParser.cc

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 1996-2025 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+/* DEBUG: section 55    HTTP Header Field Parser */
+
+#include "squid.h"
+#include "base/CharacterSet.h"
+#include "base/Raw.h"
+#include "base/TextException.h"
+#include "debug/Stream.h"
+#include "http/one/FieldParser.h"
+#include "parser/Tokenizer.h"
+#include "sbuf/Stream.h"
+#include "SquidConfig.h"
+
+/**
+ * Governed by RFC 9112 section 5:
+ *
+ *  field-line   = field-name ":" OWS field-value OWS
+ */
+void
+Http1::FieldParser::parseFieldLine(::Parser::Tokenizer &tok, const http_hdr_owner_type msgType)
+{
+    // Isolate the: field-name
+    parseFieldName(tok);
+
+    // TODO: remove when String is gone from Squid
+    if (name().length() > 65534) {
+        /* String must be LESS THAN 64K and it adds a terminating NULL */
+        debugs(55, 2, "ignoring huge header field (" <<
+               Raw("field_start", name().rawContent(), 100) <<
+               "...[skip " << name().length()-100 << " characters])");
+        throw TextException(SBuf("huge-header-name"), Here());
+    }
+
+    // handle delimiter ( BWS ":" ) or abort
+    skipColonDelimiter(tok, msgType);
+
+    // Isolate the: OWS field-value OWS
+    (void)tok.skipAll(CharacterSet::WSP); // at least an SP expected
+    parseFieldValue(tok);
+    (void)tok.skipAll(CharacterSet::WSP);
+
+    // TODO: remove when String is gone from Squid
+    if (value().length() > 65534) {
+        /* String must be LESS THAN 64K and it adds a terminating NULL */
+        throw TextException(ToSBuf("'", name(), "' header of ", value().length(), " bytes"), Here());
+    }
+
+    if (!tok.atEnd()) {
+        const auto garbage = tok.remaining();
+        const auto limit = min(garbage.length(), SBuf::size_type(100));
```

Comment:
Please move this dangerous and duplicated output limiting code to the Raw class itself, adding Raw::upTo(limit) or a similar method (where we can properly report the actual length and properly add `...` to the truncated output).

Reply:
You have asked for this before, then vetoed the PR adding the requested change. Will not repeat here.

Reply:
I assume you are talking about #466. I did not veto that PR. In fact, I [explicitly supported](https://github.com/squid-cache/squid/pull/466#issuecomment-674567590) it. Once its problems are fixed, it should be merged (and used here).